File: puzzles/medium/js/stock-exchange.js
let maxValue = -1
let maxLoss = 0

const NB_VALUES = parseInt(readline());
let inputs = readline().split(' ');
for (let i = 0; i < NB_VALUES; i++) {
    let value = parseInt(inputs[i]);
    if (value > maxValue) {
      maxValue = value;
    } else {
      let loss = maxValue - value;
      maxLoss = Math.max(maxLoss, loss);
    }
}

print(-maxLoss);

File: puzzles/medium/python3/stock_exchange.py
max_value = -1
max_loss = 0

nb_values = int(input())
for i in input().split():
    value = int(i)
    if (value > max_value):
        max_value = value
    else:
        loss = max_value - value
        max_loss = max(max_loss, loss)

print(-max_loss)
